draft-ietf-httpbis-client-cert-field.md

```diff
@@ -183,11 +183,13 @@ multiple times in a request.
 ## Client-Cert-Chain HTTP Header Field {#chain-header}
 
 In the context of a TLS terminating reverse proxy deployment, the proxy
-MAY make the certificate chain used for validation of the end-entity certificate
+MAY make the certificate chain for validation of the end-entity certificate
 available to the backend application with the Client-Cert-Chain HTTP header
-field. This field contains certificates used by the proxy to validate the
-certificate used by the client in the TLS handshake. These certificates might or
-might not have been provided by the client during the TLS handshake.
+field. This field contains certificates for validation of the end-entity
+certificate provided by the client in the TLS handshake. These certificates might or
+might not have been provided by the client during the TLS handshake, however,
```

Comment:
Can you explain your reasoning behind this "might or might not" clause?

Reply:
That text was from 0a7210f7d644dea399f0fc4d1b5119acb436d1db when @MikeBishop added the Client-Cert-Chain header and I was trying not to step on his work too much. And also to allow for the case that the proxy assembles the chain and conveys it.

Reply:
This seems problematic as it invites a query about why that might (or might not) be the case.

I would prefer to say that the proxy can add certificates to this field.  It provides the certificates in the order in which they appear in the TLS handshake, unless it has cause to provide other information.

Reply:
What about even less invitation of questions by making that last sentence this? 

"The order of the certificates is the same as the ordering in TLS (such as described in {{Section 4.4.2 of RFC8446}})."

Reply:
That would work for me, but I got the impression that you are aware of cases where that is not the case.  I don't think that there is any harm in saying "..., unless the server and proxy agree otherwise." or similar.

Reply:
Honestly don't remember what my logic was in writing that; it appears to have been from my first pass at this text and not touched in any of the feedback.

I *think* that was based on TLS 1.3 saying that "...a certificate that specifies a trust anchor MAY be omitted from the chain, provided that supported peers are known to possess any omitted certificates."  If the proxy possesses certificates the origin might not, the client might omit them from the handshake, but the proxy could append them for the origin's information.

I don't think there's harm in allowing for private agreement here, but I don't feel strongly either way.

Reply:
As I remember things, there was some disagreement when the initial text for `Cient-Cert-Chain` was added/proposed about whether the proxy would construct and validate the chain and convey the constructed chain to the origin or whether the proxy would pass through what the client sent as-is and defer validation and construction to the origin. Some of this discussion is captured in PR #1874. AFAIK most of the time the chain would look the same. But I've tried to adjust the text such that either would be okay. The sentence above about the order being the same was intended to mean only that ordering is the same but not that they are the same.

Reply:
[5fb237a](https://github.com/httpwg/http-extensions/pull/2258/commits/5fb237a7dbb030ab323710a1d254d442cd41f35d) is another pass at this that removes the "might or might not" and tries to articulate ordering.